src/libawkward/array/ListOffsetArray.cpp

```diff
@@ -2040,17 +2041,16 @@ namespace awkward {
 
       outcontent = outcontent.get()->carry(outcarry, false);
 
+      Index64 outoffsets(offsets_.length());
+      // FIXME: move to kernel
+      for (int64_t i = 0; i < offsets_.length(); i++) {
+        outoffsets.data()[i] = offsets_.data()[i] - offsets_.data()[0];
+      }
+
```

Comment:
@agoose77 - same here: a temporary fix - I left it for you to provide a proper one :-)